### PR TITLE
Add dependabot config

### PR DESCRIPTION
**Pull request by kolyshkin**
This way it can create PRs to update various dependencies (in Go modules and github workflow rules).

Cc @rhatdan

### CODE REVIEW

**Review by austinvazquez — APPROVED** [review]

**Comment by austinvazquez** [comment]
LGTM, another thread I would like to start is enabling rubberstamp approvals and auto-merge automation for minor dependency updates. Dependabot can create quite a few PRs and overwhelm maintainers. I can try a proof of concept and propose a mechanism separately from this.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Please see the documentation for all configuration options:
+# https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file
+
+version: 2
+updates:
+  # Dependencies listed in go.mod
+  - package-ecosystem: "gomod"
```
Comment (thaJeztah) [inline .github/dependabot.yml:7]: With this repository being a library, I don't think we need to update module versions, unless there's a critical reason for it in the call path. Doing so would violate go modules MVS (which requires lowest version to be specified)

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (kolyshkin) [reply .github/dependabot.yml:7]: Right. I'll remove this.

(Wonder what to do about other repos such as runc, which is a library and a binary at the same time)

**Comment by kolyshkin** [comment]
> LGTM, another thread I would like to start is enabling rubberstamp approvals and auto-merge automation for minor dependency updates. Dependabot can create quite a few PRs and overwhelm maintainers. I can try a proof of concept and propose a mechanism separately from this. Perhaps we can start with selinux and if successful apply the mechanism to other projects? cc: @kolyshkin @rhatdan wdyt?

Not a big deal for this repo, as we don't have lots of dependencies and we don't require 2 LGTMs here, so any single maintainer can merge.

OTOH I'd love some automation for github.com/opencontainers/runc (which requires two LGTMs, while we can easily do with just one for dependabot and other simple PRs, but I don't know how to do that).

File: .github/dependabot.yml (hunk 1 above)
Comment (thaJeztah) [reply .github/dependabot.yml]: Yeah, those can be the tricky ones. I think currently it's probably OK (we update main, but not release branches), although we could be more conservative if the updates we pull in bring no changes to code used in runc itself

**Review by thaJeztah — APPROVED** [review]
LGTM

updating actions sounds good to me, and shouldn't hurt. Let's see if the "daily" works well

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**